vtex/faststore
# Feat: Create default fragments

## What's the purpose of this pull request?



## How it works?

This PR is part of the extension API initiative, focused on creating default GraphQL fragments for customization in the stores. Two default fragments, 'plp' and 'pdp', were added and called in ServerCollectionPageQuery and ServerProductDetailPageQuery, respectively. 

After adding the fragments, the "yarn generate" command was used to create the types.

## How to test it?

To test this feature, you can navigate to the 'package/core' directory and execute the command 'yarn dev'. Once done, you can utilize your preferred client to run the following CURL command:

```
curl --request POST \
  --url http://localhost:3000/api/graphql \
  --header 'Content-Type: application/json' \
  --data '{"query":"fragment plp on Query {\n\tcollection(slug: $slug) {\n\t\tid\n\t}\n}\n\nquery ServerCollectionPageQuery($slug: String!) {\n\t...plp\n\tcollection(slug: $slug) {\n\t\tseo {\n\t\t\ttitle\n\t\t\tdescription\n\t\t}\n\t\tbreadcrumbList {\n\t\t\titemListElement {\n\t\t\t\titem\n\t\t\t\tname\n\t\t\t\tposition\n\t\t\t}\n\t\t}\n\t\tmeta {\n\t\t\tselectedFacets {\n\t\t\t\tkey\n\t\t\t\tvalue\n\t\t\t}\n\t\t}\n\t}\n}\n","operationName":"ServerCollectionPageQuery","variables":{"slug":"brand"}}'
```

Everything should work normally in the store navigation.

## References

[Task](https://vtex-dev.atlassian.net/browse/FS-1216)
